test/Google.Api.Gax.Grpc.Tests/MonitoringResourceBuilderTest.cs

```diff
@@ -73,17 +73,45 @@ public void GkePlatform()
     'attributes': {
       'cluster-name': 'FakeClusterName',
     },
+    'id': '123',
     'zone': 'projects/FakeProject/zones/FakeLocation'
   }
 }
 ";
-            var platform = new Platform(GkePlatformDetails.TryLoad(json));
+            const string namespaceJson = @"
+{
+  'kind': 'Namespace',
+  'metadata': {
+    'uid': 'namespaceid'
+  }
+}
+";
+            const string podJson = @"
+{
+  'kind': 'Pod',
+  'metadata': {
+    'name': 'podname',
+    'uid': 'podid'
+  }
+}
+";
+            var kubernetesData = new GkePlatformDetails.KubernetesData
+            {
+                NamespaceJson = namespaceJson,
+                PodJson = podJson,
+                MountInfo = new[] { "/var/lib/kubelet/pods/podid/containers/containername/ /dev/termination-log" }
+            };
+            var platform = new Platform(GkePlatformDetails.TryLoad(metadataJson, kubernetesData));
             var resource = MonitoredResourceBuilder.FromPlatform(platform);
-            Assert.Equal("gke_cluster", resource.Type);
-            Assert.Equal(3, resource.Labels.Count);
+            Assert.Equal("container", resource.Type);
+            Assert.Equal(7, resource.Labels.Count);
```

Comment:
Possibly build an expected dictionary and just `Assert.Equal(expectedLabels, resource.Labels)`?

Reply:
done